src/bench/pool.cpp

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) 2022 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <bench/bench.h>
+#include <support/allocators/pool.h>
+
+#include <unordered_map>
+
+template <typename Map>
+void BenchFillClearMap(benchmark::Bench& bench, Map& map)
+{
+    size_t batch_size = 5000;
+
+    // make sure each iteration of the benchmark contains exactly 5000 inserts and one clear.
+    // do this at least 10 times so we get reasonable accurate results
+
+    bench.batch(batch_size).minEpochIterations(10).run([&] {
+        auto rng = ankerl::nanobench::Rng(1234);
+        for (size_t i = 0; i < batch_size; ++i) {
+            map[rng()];
+        }
+        map.clear();
+    });
```

Comment:
This benchmark doesn't re-use memory (add to freelist then remove from freelist); maybe it would be better if it did because that's what happens with the real coins cache, maybe something like this:

<details>
<summary>patch</summary>

```diff
-    size_t batch_size = 5000;
+    // The steady-state size of the map will be half of this value;
+    // power-of-2 to avoid expensive mod operation during benchmark.
+    constexpr size_t batch_size = 1 << 13;
 
     // make sure each iteration of the benchmark contains exactly 5000 inserts and one clear.
     // do this at least 10 times so we get reasonable accurate results
 
     bench.batch(batch_size).minEpochIterations(10).run([&] {
         auto rng = ankerl::nanobench::Rng(1234);
-        for (size_t i = 0; i < batch_size; ++i) {
-            map[rng()];
+        for (size_t i = 0; i < batch_size * 10; ++i) {
+            uint64_t r{rng()};
+            // if the map has few entries, more likely add an entry, else delete
+            if ((r & (batch_size-1)) < map.size()) {
+                map.erase(map.begin());
+            } else {
+                map[r];
+            }
         }
         map.clear();
     });
```
</details>

I don't think it matters which entry we delete, since `map.begin()` could return any entry. Deleting a truly random entry gets more complicated, and I don't think it changes the performance significantly. The loop iterates `batch_size * 10` instead of `batch_size` so we reach steady-state (about equal numbers of insertions and deletions).

Reply:
Actually it does reuse memory, when calling `map.clear()` the nodes are put back into the freelist. All elements are deallocated, so he `pool_resource` gets them and puts them into their freelist. The memory is only ever released in the destructor of `pool_resource`

Reply:
That's a good point, I hadn't noticed that. But perhaps it's good to make the load more realistic to get better benchmarking results. I'm assuming that when UTXOs become spent TXOs during block validation, they're removed from this unordered_map, but maybe that's not true? Just seems like doing many interleaving allocations and deallocations would be more realistic, but maybe not. But this suggestion is non-blocking.

In case anyone is wondering, I observed that the benchmark does cause allocations (and deallocations) that are too large for the pool, and these are for the hash bucket arrays. (I'm sure the fuzzer does that too, so far I've run only the benchmark.)